### ISSUE TITLE
Can't set an initial state for the Bidirectional LSTM Layer of tf.keras 2.0 under eager execution mode

### ISSUE THREAD

**Issue by Liu-Da** [tensorflow/tensorflow#28761]
**System information**
- Have I written custom code: No
- OS Platform and Distribution: MacOS 10.14.4 
- TensorFlow installed from: pip install tensorflow==2.0.0-alpha0 
- TensorFlow version: v1.12.0-9492-g2c319fb415 2.0.0-alpha0
- Python version: 3.7.3

**Describe the current behavior**
- Get an Error when setting an initial state for the Bidirectional LSTM Layer of tf.keras 2.0 under eager execution mode

**Describe the expected behavior**
- The code should work with eager execution mode.

**Code to reproduce the issue**
```
import tensorflow as tf
import numpy as np


hidden_units = 64
time_step = 50
vocab_size = 100
embed_size = 64
batch_size = 10

embedding = tf.keras.layers.Embedding( vocab_size, embed_size, input_length=time_step, trainable=False )
lstm =  tf.keras.layers.LSTM(hidden_units, return_sequences=False, return_state=True )
bilstm_1 =  tf.keras.layers.Bidirectional(lstm, name='bilstm1')
bilstm_2 =  tf.keras.layers.Bidirectional(lstm, name='bilstm2')
concat =  tf.keras.layers.Concatenate()
dense = tf.keras.layers.Dense(vocab_size, activation='softmax')

def mod(input_1,input_2):
    
    input_1 = embedding(input_1)
    input_2 = embedding(input_2)
    output_1, forward_h, forward_c, backward_h, backward_c = bilstm_1(input_1)
    output_2, _,_,_,_ = bilstm_2(input_2, initial_state=(forward_h, forward_c, backward_h, backward_c))
    output = dense(concat([output_1,output_2]))
    return output

input_1=np.random.randint(0,99,size=[batch_size, time_step])
input_2=np.random.randint(0,99,size=[batch_size, time_step])
target = np.random.randint(2,size=[batch_size])

#test the model
mod(input_1,input_2)

```

**Other info / logs**
The Error detail：
`InvalidArgumentError: Incompatible shapes: [2,256] vs. [50,256] [Op:Add] name: bilstm2/add/`

**Comment by muddham** [comment]
@Liu-Da Able to reproduce the issue with the provided code.

**Comment by qlzh727** [comment]
Thanks for reporting the issue, will send out the fix very soon.

**Comment by qlzh727** [comment]
This should be fixed by 2a8f9b1ccfaaebd6f9cf5b5eb972c2dafded4f5e now.

**Comment by aakhundov** [comment]
@qlzh727 Same problem in the `Bidirectional` code leads to a different issue in tf 1.14 (graph mode):

**Code to reproduce the issue**

```
lstm = tf.keras.layers.LSTM(units=10)
bidirectional = tf.keras.layers.Bidirectional(lstm)

inputs = tf.placeholder(tf.float32, [32, 10, 20])
fw_state = [tf.zeros([32, 10]), tf.zeros([32, 10])]
bw_state = [tf.zeros([32, 10]), tf.zeros([32, 10])]
initial_state = fw_state + bw_state

output = bidirectional(
    inputs=inputs, initial_state=initial_state)
``` 

**Other info / logs**

```
TypeError: object of type 'Tensor' has no len()
```

Do you think this can be solved in tf 1.x, too, at some point?

Thanks!
